webapp/views.py

```diff
@@ -934,6 +944,38 @@ def get_purchase(purchase_id):
         return flask.jsonify({"error": "authentication required"}), 401
 
 
+def get_purchase_account():
+    auth_token = None
+
+    if user_info(flask.session):
+        auth_token = flask.session["authentication_token"]
+
+    advantage = AdvantageContracts(
+        session,
+        auth_token,
+        api_url=flask.current_app.config["CONTRACTS_API_URL"],
+    )
+
+    if not flask.request.is_json:
+        return flask.jsonify({"error": "JSON required"}), 400
+
+    email = flask.request.json.get("email")
+    payment_method_id = flask.request.json.get("payment_method_id")
+
+    try:
+        account = advantage.get_purchase_account(email, payment_method_id)
+
+        flask.session["guest_authentication_token"] = account["token"]
+    except HTTPError as http_error:
+        flask.current_app.extensions["sentry"].captureException()
+        return (
```

Comment:
As mentioned, this could not be always an actual 500. SOmetimes we should just suggest users to log in instead, because you can only purchase as a guest if the email that you claim is not known by ua-contracts.